### PR TITLE
Match whirlwind effect

### PR DESCRIPTION
**Pull request by pixel-stuck**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀 This PR matches 3 functions (+0.10%) on `us`.

File: src/effects/whirlwind.c [hunk 1]
```diff
@@ -1,29 +1,283 @@
...
+    } else {
+        part->unk_10 = arg5;
+    }
+    part->primAlpha = 0;
+    part->pos.x = arg1;
+    part->pos.y = arg2;
+    part->pos.z = arg3;
+    part->unk_34 = arg4;
+
+    part->primR = 0xFF;
```
Comment (ethteck) [inline src/effects/whirlwind.c:45]: decimal for the colors pleez

File: src/effects/whirlwind.c [hunk 2]
```diff
@@ -1,29 +1,283 @@
...
+    part->unk_138 += 10.0f;
+    part->unk_13C += 12.96;
+    part->unk_140 += 17.28;
+    if (part->unk_13C > 64.0f) {
+        part->unk_13C -= 64.0f;
+    }
+    if (part->unk_140 > 64.0f) {
+        part->unk_140 -= 64.0f;
+    }
+    for(i = 0; i < 8; i++) {
```
Comment (ethteck) [inline src/effects/whirlwind.c:128]: ARRAY_LENGTH(x)

File: src/effects/whirlwind.c [hunk 3]
```diff
@@ -1,29 +1,283 @@
...
-    retTask->renderMode |= RENDER_TASK_FLAG_2;
+    queuedTask = shim_queue_render_task(&renderTask);
+    queuedTask->renderMode |= RENDER_TASK_FLAG_2;
 }
 
 void func_E00CE470(void) {
 }
 
-INCLUDE_ASM(s32, "effects/whirlwind", whirlwind_appendGfx);
+void whirlwind_appendGfx(void *effect) {
```
Comment (ethteck) [inline src/effects/whirlwind.c:149]: pointer on the left

File: src/effects/whirlwind.c [hunk 4]
```diff
@@ -1,29 +1,283 @@
...
+    gSPBranchList(savedPos, gMasterGfxPos);
+    gSPClearGeometryMode(gMasterGfxPos++, G_CULL_BOTH);
+    gDPSetPrimColor(gMasterGfxPos++, 0, 0, data->primR, data->primG, data->primB, primAlpha);
+    gDPSetEnvColor(gMasterGfxPos++, data->envR, data->envG, data->envB, 32);
+
+    // Call the main display list
+    gSPDisplayList(gMasterGfxPos++, whirlwindMainDisplayList);
+    gSPPopMatrix(gMasterGfxPos++, G_MTX_MODELVIEW);
+    gSPPopMatrix(gMasterGfxPos++, G_MTX_MODELVIEW);
+}
```
Comment (ethteck) [inline src/effects/whirlwind.c:283]: empty line at end of file plez

**Review by ethteck — CHANGES_REQUESTED** [review]
nish!

**Review by nanaian — APPROVED** [review]

**Pull request merged**